<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<HTML>
<HEAD>

<META NAME="description" CONTENT="Estimation of population parameters using genetic data usi
ng a maximum likelihood approach with Metropolis-Hastings Monte Carlo Markov chain importanc
e sampling">
<META NAME="keywords" CONTENT="MCMC, Markov chain, Monte Carlo, Metropolis-Hastings, populat
ion, parameters, migration rate, population size, recombination rate, maximum likelihood">

<TITLE>LAMARC Documentation</title>
</HEAD>

<BODY BGCULOR="#FFFFFF" TEXT="#000000">
<!-- coalescent, coalescence, Markov chain Monte Carlo simulation, migration rate, effective
 population size, recombination rate, maximum likelihood -->
<H1 ALIGN="CENTER"> LAMARC - Likelihood Analysis with Metropolis Algorithm
using Random Coalescence<P>
<IMG SRC="images/lamarc.gif" BORDER="1"  HEIGHT="100" ALT="[LAMARC logo]" 
ALIGN="MIDDLE"></P>
</H1>

<P> This documentation covers use of the program LAMARC, which can estimate
population sizes, migration rates, growth rates, and recombination rate
using a Metropolis-Hastings Monte Carlo Markov Chain algorithm.  LAMARC's
'sister program' is <A HREF="http://popgen.sc.fsu.edu/">MIGRATE</a>, with
which it shares history, algorithms, and some code base, but which is now
being maintained exclusively by Peter Beerli at Florida State
University.</P>

<UL>
<H3>General:</H3>
<LI> <A HREF="overview.html">Overview</A></LI>
<LI> <A HREF="changes.html">Changes in Version 2.1.9</A></LI>
<LI> <A HREF="upcoming.html">Plans for the Future</A></LI>
<LI> <A HREF="troubleshooting.html">Frequently Asked Questions</A></LI>
<LI> <A HREF="messages.html">Error and Warning Messages</A></LI>
<LI> <A HREF="limitations.html">Limitations of LAMARC</A></LI>
<LI> <A HREF="glossary.html">Glossary of Terms</A></LI>
<H3>Tutorials:</H3>
<LI> <A HREF="data_required.html">What kind of data to use with LAMARC</A></LI>
<LI> <A HREF="tutorial.html">How to design a LAMARC analysis</A></LI>
<LI> <A HREF="bayes_howto.html">How to design a Bayesian LAMARC analysis</A></LI>
<LI> <A HREF="tutorial2.html">Interpreting the results</A></LI>
<LI> <A HREF="parallel.html">Combining multiple computers to complete a LAMARC run</A></LI>
<LI> <A HREF="mapping.html">Using LAMARC for fine-scale mapping</A></LI>
<H3>Specific Topics:</H3>
<LI> <A HREF="compiling.html">Compiling LAMARC</A></LI>
<LI> <A HREF="converter.html">File conversion utilities</A></LI>
<LI> <A HREF="genetic_map.html">Modeling Linkage Properties and Relative Mutation Rates</A></LI>
<LI> <A HREF="migration_matrix.html">Migration Matrix</A></LI>
<LI> <A HREF="divergence.html">Divergence</A></LI>
<LI> <A HREF="panels.html">Panel Corrections</A></LI>
<LI> <A HREF="converter_cmd.html">The Converter Command File</A></LI>
<LI> <A HREF="xmlinput.html">XML data input format</A></LI>
<LI> <A HREF="menu.html">Using the LAMARC interactive menu</A></LI>
<LI> <A HREF="regions.html">Dividing your data into regions and segments</A></LI>
<LI> <A HREF="data_models.html">Modelling the mutational process</A></LI>
<LI> <A HREF="gamma.html">Handling variation in mutation rate (several types, including gamma-distributed rates)</A></LI>
<LI> <A HREF="forces.html">Evolutionary forces</A></LI>
<LI> <A HREF="parameters.html">What do the parameters mean?</A></LI>
<LI> <A HREF="search.html">Search strategies</A></LI>
<LI> <A HREF="output.html">Interpreting the output</A></LI>
<LI> <A HREF="bayes.html">Bayesian analysis</A></LI>
<LI> <A HREF="tracer.html">Using Tracer to examine LAMARC results</A></LI>
<LI> <A HREF="genotype.html">Genotypic (phase-unknown) data</A></LI>
<LI> <A HREF="viral_data.html">Specific issues with virus data</A></LI>
</UL>
<H3>References:</H3>
<P><A HREF="http://www.genetics.org/cgi/content/abstract/175/1/155">Kuhner,
M. K. and L. P. Smith, 2007  <i>"Comparing Likelihood and Bayesian Coalescent
Estimation of Population Parameters"</i> Genetics 175: 155-165.</a></P>
<P>

<P><A
HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/22/6/768">
Kuhner, M. K., 2006  <i>"LAMARC 2.0: maximum likelihood and Bayesian estimation  of
population parameters."</i>  Bioinformatics 22(6): 768-770.</a> </P>


<A HREF="overview.html">Next</A>
<HR>
Please report problems to <A HREF="MAILTO:lamarc@u.washington.edu">
lamarc@u.washington.edu</A>. The LAMARC package is maintained by the
lab of
<A HREF="http://evolution.gs.washington.edu/lamarc/mkkuhner.html"> Mary K. Kuhner </A>
<ADDRESS> Mary Kuhner,
Department of Genome Sciences,
University of Washington, Box 355065,
Seattle, WA  98195-5065,  USA.  Phone: (206) 543-8751, Fax: (206) 543-0754</ADDRESS>

<!--
//$Id: index.html,v 1.51 2013/11/08 23:09:53 mkkuhner Exp $
-->
</BODY>
</HTML>

