<!-- header fragment for html documentation -->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<HTML>
<HEAD>

<META NAME="description" CONTENT="Estimation of population parameters using genetic data using a maximum likelihood approach with Metropolis-Hastings Monte Carlo Markov chain importance sampling">
<META NAME="keywords" CONTENT="MCMC, Markov chain, Monte Carlo, Metropolis-Hastings, population, parameters, migration rate, population size, recombination rate, maximum likelihood">

<TITLE>LAMARC Documentation: Overview</title>
</HEAD>


<BODY BGCOLOR="#FFFFFF">
<!-- coalescent, coalescence, Metropolis-Hastings, Markov chain Monte Carlo
 simulation, migration rate, effective population size, recombination rate,
 maximum likelihood -->


(<A HREF="index.html">Contents</A> | <A HREF="changes.html">Next</A>) 

<H2>Overview of LAMARC </H2>

<p> LAMARC is a large, complex, powerful set of data analysis tools to do coalescence of populations. It can calculate a wide variety of forces and use a large number of different methods. Because of all its options it is not easy to learn. We have done our best to cover the topics that most people need in this web site, but you need to both read the documents and be willing to experiment with various methods to find what works for your data. We recommend that you read about LAMARC's strengths and limits before you start collecting data. We also realize this is unlikely to happen. <A HREF="MAILTO:lamarc@u.washington.edu">We</a> will be glad to help as much as we can, both with experimental design and how to do data analysis, but, ultimately, you are the expert on your data and know what you need. We can only help you with how you ask your questions.</p>

<H2>Family tree</H2>

<P> All versions of LAMARC combine much of the capabilities of the previous
programs MIGRATE, RECOMBINE, and FLUCTUATE, and each has added new
functionality as well.  Using DNA or RNA sequence data, SNPs,
microsatellites, or K-Allele data such as electrophoretic alleles, LAMARC
can estimate Theta, recombination rate, migration rates, and growth (or
decline) rates for the population(s) from which the data were drawn.  These
four forces can be estimated all together or with any sub-combination that
includes Theta (i.e. Theta plus up to three other forces).  It can also make
estimates using genotyped (or unphased) data.</P>

<P>As of version 2.1, LAMARC can now be used for fine-scale mapping of trait
data.  Recombination is required for this capability, which can be performed
in the presence or absence of migration or growth.</P>

<P>The older programs RECOMBINE and FLUCTUATE are no longer being actively
supported, as their capabilities have been incorporated into LAMARC.  <A
HREF="http://popgen.sc.fsu.edu/">MIGRATE</a> is now being maintained by
Peter Beerli at Florida State University.  He has
taken the program in a slightly different direction than LAMARC, and it now
offers unique features such as the ability to run the program in parallel on
a cluster, and various diverse models for migration.  </P>

<P>The primary advantage of LAMARC over the older programs is its ability to
simultaneously estimate what the other programs estimated separately.  Even
if you are primarily interested in only one of these, their simultaneous
estimation means that your estimates will not be biased by the
unacknowledged presence of the other.</P>

<P>LAMARC is written in C++.  Each release includes executables which 
should run on current versions of Linux, OS X, and MS Windows.
For more information see
<a href="compiling.html">Compiling Lamarc</a>.</p>

<P>The program is free to download and use.  We would appreciate
hearing about any publications resulting from it. To cite LAMARC,
you can reference our announcement paper:</P>

<P><A
HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/22/6/768">
Kuhner, M. K., 2006  <i>"LAMARC 2.0: maximum likelihood and Bayesian estimation  of
population parameters."</i>  Bioinformatics 22(6): 768-770.</a> </P>

<P>For more information about the Bayesian aspects of the program, see:</P>

<P><A HREF="http://www.genetics.org/cgi/content/abstract/175/1/155">Kuhner,
M. K. and L. P. Smith, 2007  <i>"Comparing Likelihood and Bayesian Coalescent
Estimation of Population Parameters"</i> Genetics 175: 155-165.</a></P>
<P>

<P> Bug reports, comments, critiques, and notices of papers can be
sent to <A HREF="mailto:lamarc@u.washington.edu">lamarc@u.washington.edu</A>. </P>

<P>The program can be found for download on our Web site:</P>

<A HREF="http://evolution.gs.washington.edu/lamarc/">
http://evolution.gs.washington.edu/lamarc/</A>

<H4>About the Authors:</H4>

<P>The LAMARC program is currently being developed at the University of
Washington in the Felsenstein/Kuhner lab.  The list of contributors
includes:</P>

    <UL>
      <LI> Mary K. Kuhner  (UW, current development team)
      <LI> Jon Yamato (UW, current)
      <LI> Bob Giansiracusa (UW, current)
      <LI> Jim McGill (UW, current)
      <LI> Elizabeth Walkup (UW, current)
      <LI> Peter Beerli (FSU, former contributor, now working on MIGRATE)
      <LI> Patrick Colacurcio (UW, former contributor)
      <LI> Chia-Chi Li (UW, former)
      <LI> Eric Rynes (UW, former)
      <LI> Jim Sloan (UW, former)
      <LI> Lucian Smith (UW, former)
      <LI> Wang Yi (UW, former)
    </UL>

<P> Other people and organizations who have provided essential infrastructure
support for this project:
<ul>
<LI> <A HREF="http://www.boost.org/libs/smart_ptr/smart_ptr.htm">Boost</a> for providing the smart_ptr implementation
<LI> <A HREF="http://www.gnu.org/">GNU</a> for the g++ compiler, gdb debugger, and CVS source control 
system used to develop this program.  
<LI> Lee Thomason and Yves Berquin for their <A
HREF="http://sourceforge.net/projects/tinyxml">TinyXML parser</A>.
<LI> <A HREF="http://www.wxwidgets.org/">wxWidgets</a> for the tools to make
the new GUI for our file converter.
<LI> Matthew Austern in particular and the denizens of the
usenet newsgroup <A
HREF="http://groups.google.com/group/comp.lang.c++.moderated/">
comp.lang.c++.moderated</a> in general for helpful advice.
</ul>
</P>
<P>Funding for this project was provided by the National Institutes
of Health grants GM 51929-02 and HG 01989-02, both to Joseph
Felsenstein, and the NIH grant GM 51929-10 to Mary K. Kuhner.</P>

(<A HREF="index.html">Contents</A> | <A HREF="changes.html">Next</A>)
<!--
//$Id: overview.html,v 1.35 2012/05/14 19:55:38 ewalkup Exp $
-->
</BODY>
</HTML>
